test/javascripts/acceptance/chat-test.js

```diff
@@ -1225,6 +1227,32 @@ acceptance("Discourse Chat - chat preferences", function (needs) {
     assert.equal(currentURL(), "/u/eviltrout/preferences/chat");
     assert.equal(queryAll(".chat-setting").length, 4);
   });
+
+  test("The user can save the settings", async function (assert) {
+    updateCurrentUser({ has_chat_enabled: false });
+    let spy = sinon.spy(ajaxlib, "ajax");
+    await visit("/u/eviltrout/preferences/chat");
+    await click("#user_chat_enabled");
+    await click("#user_chat_only_push_notifications");
+    await click("#user_chat_isolated");
+    await selectKit("#user_chat_sounds").expand();
+    await selectKit("#user_chat_sounds").selectRowByValue("bell");
+
+    await click(".save-changes");
+
+    assert.ok(
+      spy.calledWithMatch("/u/eviltrout.json", {
```

Comment:
Usually I would say we want to avoid mocks but I have no idea how else we can test this properly :man_shrugging:

Reply:
@tgxworld yeah it's the same in core, have to use sinon to see what is actually going to get sent to the server. `spy` at least is just a wrapper around the original method, doesn't mock it out completely so that's nice